### PR TITLE
Merging feature/output into master

### PR DESCRIPTION
**Pull request by tomdng**
Finally got the drone output and basic error detection working! This feature took a bit longer than usual because it required that a lot of the other stuff was working first (like the text editor, tutorial selection, and etc).

Henri built out the basic shell of the output back when Propguard was a really small project and then I went and fleshed it out a lot more including adding all the logic into it once we redesigned the entire interface.

Resolves features #4 and #14 .

### CODE REVIEW

File: src/components/navbar/index.tsx [hunk 1]
```diff
@@ -51,25 +51,36 @@ const NavIcon: AnyStyledComponent = styled(Icon)`
...
 
-  const toggleDisplay = (): void => {
+  const toggleDisplay = (display: boolean): void => {
+    if (display) {
+      setSelectorDisplay(true);
+      setTutorialDisplay(false);
+    } else {
+      setSelectorDisplay(false);
+      setTutorialDisplay(true);
+    }
```
Comment (ClayMav) [inline src/components/navbar/index.tsx]: Will these two values always bee opposites?

File: src/components/navbar/index.tsx [hunk 2]
```diff
@@ -51,25 +51,36 @@ const NavIcon: AnyStyledComponent = styled(Icon)`
...
       setSelectorView("true");
     }
-    setSelectorDisplay(!selectorDisplay);
-    setTutorialDisplay(!tutorialDisplay);
   };
 
   return (
     <NavWrapper>
-      <NavLogo to="/" onClick={toggleDisplay}>
+      <NavLogo to="/" onClick={(): void => toggleDisplay(false)}>
```
Comment (ClayMav) [inline src/components/navbar/index.tsx]: lambdas are slow in react renders, statically define this as a function

File: src/components/navbar/index.tsx [hunk 3]
```diff
@@ -51,25 +51,36 @@ const NavIcon: AnyStyledComponent = styled(Icon)`
...
         <Logo />
       </NavLogo>
-      <NavIcon icon={listIcon} width="2.5rem" onClick={toggleDisplay} />
+      <NavIcon
+        icon={listIcon}
+        width="2.5rem"
+        onClick={(): void => toggleDisplay(true)}
```
Comment (ClayMav) [inline src/components/navbar/index.tsx]: onClick lambda bad - as above

**Review by ClayMav — CHANGES_REQUESTED** [review]
On windows, this tool doesn't work properly due to the carriage return, or maybe something else. Newlines and spacing should not play much a role if any.
The right column expands horizontally creating a scrollbar when the output line goes past a certain length, this should not happen.
Also the listed altitude does not match the altitude of the code or the "output", same for velocity.
The hint button currently seems confusing in terms of its functionality

**Comment by tomdng** [comment]
Whoops, I've went ahead and fixed the issue for windows. Turns out there's a lot of whitespace dependent stuff I've been doing for the error checking. Now the only whitespace that the answer relies on should only be at the beginning of a line of code because of Python.

I've worked out the other issues, and since I'm planning to add the tutorial editor to Propguard, I've stopped using booleans to toggle the tutorial views and instead am using a string to select a component to display.

**Comment by tomdng** [comment]
I also forgot, the hint functionality isn’t an official feature in this PR. In the future I want it to display multiple hints and have an option for filling in the answer without using a browser alert window.

File: src/components/tutorialComponent/index.tsx [hunk 4]
```diff
@@ -121,25 +122,20 @@ const TutorialDisplay: React.FC = (): JSX.Element => {
...
+  // componentView and whether or not it equals "TutorialComponent"
   return (
-    <TutorialBG disp={tutorialDisplay}>
+    <TutorialBG disp={componentView}>
       <TutorialHeader>
         <StepButton
-          clickFunction={(): Promise<{ tutorialStep: number }> =>
-            handleStep(false)
-          }
+          clickFunction={(): void => handleStep(false)}
```
Comment (ClayMav) [inline src/components/tutorialComponent/index.tsx]: lambda :(

File: src/components/tutorialComponent/index.tsx [hunk 5]
```diff
@@ -121,25 +122,20 @@ const TutorialDisplay: React.FC = (): JSX.Element => {
...
           next={false}
           tutorialStep={tutStep}
           totalSteps={data.instructions.length}
         />
         <h1>{data.instructions[tutStep - 1].title}</h1>
         <StepButton
-          clickFunction={(): Promise<{ tutorialStep: number }> =>
-            handleStep(true)
-          }
+          clickFunction={(): void => handleStep(true)}
```
Comment (ClayMav) [inline src/components/tutorialComponent/index.tsx]: lambda :(

**Review by ClayMav — APPROVED** [review]
I'd change the lambdas but it looks alright to me

**Comment by tomdng** [comment]
Oh shoot my bad, I'll make sure to get rid of those lambda functions. Thanks for the PR review!

**Pull request merged**